lib/components/data/hooks/useNwpsGaugeData.ts

```diff
@@ -11,7 +12,17 @@ const getNwpsGaugeData = async (sid, product) => {
   return response.json();
 };
 
-const useNwpsGaugeData = ({ sid, product, queryOptions }) => {
+interface useNwpsGaugeDataParams {
```

Comment:
Should we move all the interfaces to an interfaces directory? What is common practice for them?

Assuming the best option isn't to put it on the github types repo

Reply:
OOo because we could use the interfaces to build the type declarations / param lists for the docs

Reply:
I've heard of it being done both ways.  I think typically "best practice" is to keep interfaces with the relevant objects unless you have a need to access it outside of the file.  And at that point you can either export it or move it to a separate file.  I tend to prefer keeping them together so I usually just export it at that point, but if we were wanting to export _everything_ for auto-doc purposes then that might require a little more thinking.

I don't think there'd be all that much to gain by splitting it out to the types repo.  Would add a lot of maintenance overhead with a pretty negligible size reduction if you're only using JS.